src/predict.rs

```diff
@@ -426,11 +431,51 @@ pub trait Intra: Dim {
     }
   }
 
+  #[target_feature(enable = "ssse3")]
+  #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+  unsafe fn pred_cfl_ssse3(
+    output: &mut [u16], stride: usize, ac: &[i16], alpha: i16,
+    bit_depth: usize
+  ) {
+    let alpha_sign = _mm_set1_epi16(alpha);
+    let alpha_q12 = _mm_slli_epi16(_mm_abs_epi16(alpha_sign), 9);
+    let dc_q0 = _mm_set1_epi16(output[0] as i16);
+    let max = _mm_set1_epi16((1 << bit_depth) - 1);
+
+    for (line, luma) in
+      output.chunks_mut(stride).take(Self::H).zip(ac.chunks(32))
+    {
+      let mut i = 0isize;
+      while (i as usize) < Self::W {
```

Comment:
What happens if you unroll it 2x? (the fact that using the iterator does not unroll it might or might not be a bug.

Reply:
This already fully unrolls for 4x4. I'll update this once more before we merge. Using pointer arithmetic for the outer loop eliminates branching.